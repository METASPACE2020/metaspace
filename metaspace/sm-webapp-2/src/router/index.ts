import { createRouter, createWebHistory } from 'vue-router'
import { Component } from 'vue'
import NotFound from '../modules/App/NotFoundPage.vue'
import AboutPage from '../modules/App/AboutPage'
import DatasetsPage from '../modules/Datasets/DatasetsPage.vue'

const asyncPagesFreelyTyped = {
  AnnotationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "AnnotationsPage" */ '../modules/Annotations/AnnotationsPage.vue'),
  MetadataEditPage: () => import(/* webpackPrefetch: true, webpackChunkName: "MetadataEditPage" */ '../modules/MetadataEditor/MetadataEditPage.vue'),
  UploadPage: () => import(/* webpackPrefetch: true, webpackChunkName: "UploadPage" */ '../modules/MetadataEditor/UploadPage.vue'),
  DatasetOverviewPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ '../modules/Datasets/overview/DatasetOverviewPage'),
  DatasetComparisonPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ '../modules/Datasets/comparison/DatasetComparisonPage'),
  DatasetBrowserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ '../modules/Datasets/imzml/DatasetBrowserPage'),

  // These pages are relatively small as they don't have any big 3rd party dependencies, so pack them together
  DatasetTable: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/Datasets/list/DatasetTable.vue'),
  HelpPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/App/HelpPage.vue'),
  CreateGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/Group/CreateGroupPage.vue'),
  ProjectsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/Project/ProjectsListPage.vue'),
  GroupsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/Group/GroupsListPage'),
  PrivacyPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/App/PrivacyPage.vue'),

  // These pages use sanitizeHtml, which is big
  ViewGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ '../modules/Group/ViewGroupPage.vue'),
  ViewProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ '../modules/Project/ViewProjectPage.vue'),

}
const asyncPages = asyncPagesFreelyTyped as Record<keyof typeof asyncPagesFreelyTyped, Component>

export const routes : any =[
  { path: '/', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
  { path: '/about', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
  { path: '/annotations', name: 'annotations', component: asyncPages.AnnotationsPage },
  {
    path: '/datasets',
    component: DatasetsPage,
    children: [
      { path: '', component: asyncPages.DatasetTable },
      { path: 'summary', component: asyncPages.DatasetTable },
    ],
  },
  {
    path: '/datasets/:dataset_id/comparison',
    name: 'datasets-comparison',
    component: asyncPages.DatasetComparisonPage,
  },
  { path: '/datasets/edit/:dataset_id', name: 'edit-metadata', component: asyncPages.MetadataEditPage },
  { path: '/datasets/:dataset_id/add-optical-image', name: 'add-optical-image', component: asyncPages.DatasetTable },
  { path: '/dataset/:dataset_id', name: 'dataset-overview', component: asyncPages.DatasetOverviewPage },
  { path: '/dataset/:dataset_id/annotations', name: 'dataset-annotations', component: asyncPages.AnnotationsPage },
  { path: '/dataset/:dataset_id/browser', name: 'dataset-browser', component: asyncPages.DatasetBrowserPage },
  { path: '/dataset/:dataset_id/enrichment', name: 'dataset-enrichment', component: asyncPages.DatasetTable },
  { path: '/upload', component: asyncPages.UploadPage },
  { path: '/help', component: asyncPages.HelpPage, meta: { footer: true } },



  { path: '/groups', component: asyncPages.GroupsListPage },
  { path: '/group/create', component: asyncPages.CreateGroupPage },
  { path: '/group/:groupIdOrSlug', name: 'group', component: asyncPages.ViewGroupPage },


  { path: '/project/:projectIdOrSlug', name: 'project', component: asyncPages.ViewProjectPage },
  { // Legacy URL sent in "request access" emails up until Feb 2019
    path: '/project/:projectIdOrSlug/manage',
    redirect: { path: '/project/:projectIdOrSlug', query: { tab: 'members' } },
  },
  { path: '/projects', component: asyncPages.ProjectsListPage },



  { path: '/privacy', component: asyncPages.PrivacyPage, meta: { footer: true } },


  {
    path: '/:pathMatch(.*)*',
    component: NotFound,
  },
]

const router = createRouter({
  // @ts-ignore
  history: createWebHistory(import.meta.env.BASE_URL),
  routes,
  scrollBehavior(to, from, savedPosition) {
    // return desired position
    if (savedPosition) {
      return savedPosition;
    }
    return { top: 0 };
  },
})

const pageLoadedAt = Date.now()
router.beforeEach((to, from, next) => {
  // If user has had the window open for a long time, it's likely they're using old code. This can cause issues
  // such as incorrect API usage, or causing error reports for bugs that have already been fixed.
  // Force the browser to reload the page, preferring to do so during page transitions to minimize impact.
  const daysSincePageLoad = (Date.now() - pageLoadedAt) / 1000 / 60 / 60 / 24
  if (to.path !== from.path && daysSincePageLoad > 1.75 || daysSincePageLoad > 7) {
    window.location.assign(to.fullPath)
    next(false)
  } else {
    next()
  }
})


const { href } = router.resolve({ name: 'project', params: { projectIdOrSlug: 'REMOVE' } }, undefined)
export const PROJECT_URL_PREFIX = location.origin + href.replace('REMOVE', '')

export default router
