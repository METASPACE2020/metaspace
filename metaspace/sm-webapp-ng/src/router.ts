import { createRouter, createWebHistory } from 'vue-router'
import HomeView from './views/HomeView.vue'
import AboutView from './views/AboutView.vue'
import store from './store'
// ... other imports ...

const routes = [
  {
    path: '/',
    component: HomeView
  },
  {
    path: '/user/:id',
    component: AboutView
  }
]

const router = createRouter({
  history: createWebHistory(),
  routes
})

router.afterEach((to) => {
  console.log('to', to)
  // store.commit('updateFilterOnNavigate', to)
})

export default router
