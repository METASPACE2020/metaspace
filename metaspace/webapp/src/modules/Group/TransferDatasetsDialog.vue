<template>
  <el-dialog
    class="dialog"
    :model-value="true"
    append-to-body
    :title="allDatasets.length > 0 ? 'Transfer datasets' : 'Join group'"
    @close="handleClose"
  >
    <div v-loading="loading">
      <div v-if="allDatasets.length > 0">
        <p style="margin-top: 0">
          You have previously uploaded one or more datasets without a group. Do you want to transfer any of these
          datasets to {{ groupName }}?
        </p>
        <dataset-checkbox-list v-model="selectedDatasets" :datasets="allDatasets" :init-select-all="true" />
        <p v-if="!isInvited">An email will be sent to the group's principal investigator to confirm your access.</p>
        <div class="button-bar">
          <el-button :disabled="isSubmitting" @click="handleClose"> Cancel </el-button>
          <el-button type="primary" :loading="isSubmitting" @click="handleAccept">
            {{ acceptText }}
          </el-button>
        </div>
      </div>
      <div v-else>
        <p v-if="!isInvited" style="margin-top: 0">
          An email will be sent to the group's principal investigator to confirm your access.
        </p>
        <p v-else style="margin-top: 0">Are you sure you want to join {{ groupName }}?</p>
        <div class="button-bar">
          <el-button :disabled="isSubmitting" size="small" @click="handleClose"> Cancel </el-button>
          <el-button type="primary" :loading="isSubmitting" size="small" @click="handleAccept">
            {{ acceptText }}
          </el-button>
        </div>
      </div>
    </div>
  </el-dialog>
</template>

<script lang="ts">
import { defineComponent, ref, computed } from 'vue'
import { useQuery } from '@vue/apollo-composable'
import { DatasetListItem, datasetListItemsQuery } from '../../api/dataset'
import DatasetCheckboxList from '../../components/DatasetCheckboxList.vue'
import { currentUserIdQuery } from '../../api/user'
import { ElDialog, ElButton, ElLoading } from '../../lib/element-plus'

export default defineComponent({
  components: {
    DatasetCheckboxList,
    ElDialog,
    ElButton,
  },
  directives: {
    loading: ElLoading.directive,
  },
  props: {
    groupName: {
      type: String,
      required: true,
    },
    isInvited: {
      type: Boolean,
      required: true,
    },
  },
  setup(props, { emit }) {
    const isSubmitting = ref(false)
    const selectedDatasets = ref({})

    const { result: currentUserResult, loading: currenUserLoading } = useQuery(currentUserIdQuery, null, {
      fetchPolicy: 'cache-first',
    })
    const currentUser = computed(() => currentUserResult.value?.currentUser)
    const { result: allDatasetsResult, loading: datasetsLoading } = useQuery(
      datasetListItemsQuery,
      () => ({
        dFilter: {
          submitter: currentUser.value ? currentUser.value.id : null,
          hasGroup: false,
        },
      }),
      {
        enabled: computed(() => !!currentUser.value && !!currentUser.value.id),
      }
    )
    const allDatasets = computed(() => (allDatasetsResult.value?.allDatasets as DatasetListItem[]) || [])
    const loading = computed(() => currenUserLoading.value || datasetsLoading.value)
    const dialogTitle = computed(() => (allDatasets.value.length > 0 ? 'Transfer datasets' : 'Join group'))
    const acceptText = computed(() => {
      const action = props.isInvited ? 'Join group' : 'Request access'
      const numSelected = Object.values(selectedDatasets.value).filter((selected) => selected).length
      return numSelected === 0
        ? action
        : `${action} and transfer ${numSelected} ${numSelected === 1 ? 'dataset' : 'datasets'}`
    })

    const handleAccept = () => {
      const selectedDatasetIds = Object.keys(selectedDatasets.value).filter((key) => selectedDatasets.value[key])
      emit('accept', selectedDatasetIds)
      isSubmitting.value = true
    }

    const handleClose = () => {
      if (!isSubmitting.value) {
        emit('close')
      }
    }

    return {
      loading,
      isSubmitting,
      selectedDatasets,
      currentUser,
      allDatasets,
      dialogTitle,
      acceptText,
      handleAccept,
      handleClose,
    }
  },
})
</script>
<style scoped lang="scss">
.dialog ::v-deep(.el-dialog) {
  @apply max-w-lg;
}
.button-bar {
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-end;
  margin-top: 20px;
}
</style>
