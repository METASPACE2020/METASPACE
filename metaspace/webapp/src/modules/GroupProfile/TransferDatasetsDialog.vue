<template>
    <el-dialog class="dialog"
               visible
               :title="allDatasets.length > 0 ? 'Transfer datasets' : 'Join group'"
               @close="handleClose">
      <div v-loading="loading">
        <div v-if="allDatasets.length > 0">
          <p style="margin-top: 0">
            You have previously uploaded one or more datasets without a group. Do you want to transfer any of these datasets to {{groupName}}?
          </p>
          <div class="dataset-list">
            <div v-for="dataset in allDatasets">
              <el-checkbox v-model="selectedDatasets[dataset.id]">
                {{dataset.name}}
                <span class="reset-color">(Submitted {{ formatDate(dataset.uploadDT) }})</span>
              </el-checkbox>
            </div>
          </div>
          <div class="select-buttons">
            <a href="#" @click.prevent="handleSelectNone">Select none</a>
            <span> | </span>
            <a href="#" @click.prevent="handleSelectAll">Select all</a>
          </div>
          <p v-if="!isInvited">
            An email will be sent to the group's principal investigator to confirm your access.
          </p>
          <div class="button-bar">
            <el-button :disabled="isSubmitting" @click="handleClose">Cancel</el-button>
            <el-button type="primary" :loading="isSubmitting" @click="handleAccept">{{acceptText}}</el-button>
          </div>
        </div>
        <div v-else>
          <p v-if="!isInvited" style="margin-top: 0">
            An email will be sent to the group's principal investigator to confirm your access.
          </p>
          <p v-else style="margin-top: 0">
            Are you sure you want to join {{groupName}}?
          </p>
          <div class="button-bar">
            <el-button :disabled="isSubmitting" @click="handleClose">Cancel</el-button>
            <el-button type="primary" :loading="isSubmitting" @click="handleAccept">{{acceptText}}</el-button>
          </div>
        </div>
      </div>
    </el-dialog>
</template>
<script lang="ts">
  import Vue from 'vue';
  import { Component, Prop, Watch } from 'vue-property-decorator';
  import { DatasetListItem, datasetListItemsQuery } from '../../api/dataset';
  import { fromPairs } from 'lodash-es';
  import format from 'date-fns/format';

  @Component({
    apollo: {
      allDatasets: {
        query: datasetListItemsQuery,
        loadingKey: 'loading',
        variables(this: TransferDatasetsDialog) {
          return {
            dFilter: {
              submitter: this.currentUserId,
              hasGroup: false,
            }
          };
        }
      }
    }
  })
  export default class TransferDatasetsDialog extends Vue {
    @Prop({ required: true })
    currentUserId!: string; //TODO: It's weird to have this as a prop. It would be better to grab it from Vuex.
    @Prop({ required: true })
    groupName!: string;
    @Prop({ required: true })
    isInvited!: boolean; // True to show "Accept/reject invitation", false to show "Request access/cancel"

    loading: number = 0;
    allDatasets: DatasetListItem[] = [];
    selectedDatasets: Record<string, boolean> = {};
    isSubmitting: Boolean = false;

    get numSelected() {
      return Object.values(this.selectedDatasets).filter(selected => selected).length;
    }

    get acceptText() {
      const action = this.isInvited ? 'Join group' : 'Request access';
      return this.numSelected === 0
        ? action
        : `${action} and transfer ${this.numSelected} ${this.numSelected === 1 ? 'dataset' : 'datasets'}`
    }

    formatDate(date: string) {
      return `${format(date, 'YYYY-MM-DD')} at ${format(date, 'HH:mm')}`;
    }

    @Watch('allDatasets')
    populateSelectedDatasets() {
      //Rebuild `selectedDatasets` so that the keys are in sync with the ids from `datasets`
      this.selectedDatasets = fromPairs(this.allDatasets.map(({id}) => {
        return [id, id in this.selectedDatasets ? this.selectedDatasets[id] : true];
      }));
    }

    handleClose() {
      if (!this.isSubmitting) {
        this.$emit('close');
      }
    }

    handleSelectNone() {
      Object.keys(this.selectedDatasets).forEach(key => this.selectedDatasets[key] = false);
    }

    handleSelectAll() {
      Object.keys(this.selectedDatasets).forEach(key => this.selectedDatasets[key] = true);
    }

    handleAccept() {
      const selectedDatasetIds = Object.keys(this.selectedDatasets).filter(key => this.selectedDatasets[key]);
      this.$emit('accept', selectedDatasetIds);
      this.isSubmitting = true;
    }
  }
</script>
<style scoped lang="scss">
  @import "~element-ui/packages/theme-chalk/src/common/var";
  .dialog /deep/ .el-dialog {
    max-width: 600px;
  }
  .dataset-list {
    margin: 20px;
    max-height: 50vh;
    overflow: auto;
  }
  .select-buttons {
    text-align: right;
    margin: 20px;
  }
  .button-bar {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-end;
    margin-top: 20px;
  }
  .reset-color {
    color: $--color-text-regular;
  }

</style>
