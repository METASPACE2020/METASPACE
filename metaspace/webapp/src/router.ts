/* eslint-disable vue/max-len */
import Vue, { AsyncComponent } from 'vue'
import VueRouter, { RawLocation } from 'vue-router'
import AboutPage from './modules/App/AboutPage.vue'
import DatasetsPage from './modules/Datasets/DatasetsPage.vue'
import { DialogPage, ResetPasswordPage } from './modules/Account'
import { redirectAfterSignIn } from './modules/Account/signInReturnUrl'
import { updateDBParam } from './modules/Filters/url'
import NotFoundPage from './modules/App/NotFoundPage.vue'

Vue.use(VueRouter)

const asyncPagesFreelyTyped = {
  AnnotationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "AnnotationsPage" */ './modules/Annotations/AnnotationsPage.vue'),
  DatasetSummary: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetSummary" */ './modules/Datasets/summary/DatasetSummary.vue'),
  MetadataEditPage: () => import(/* webpackPrefetch: true, webpackChunkName: "MetadataEditPage" */ './modules/MetadataEditor/MetadataEditPage.vue'),
  ImageAlignmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "ImageAlignmentPage" */ './modules/ImageAlignment/ImageAlignmentPage.vue'),
  UploadPage: () => import(/* webpackPrefetch: true, webpackChunkName: "UploadPage" */ './modules/MetadataEditor/UploadPage.vue'),
  DesignPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DesignPage" */ './design/DesignPage.vue'),

  // These pages are relatively small as they don't have any big 3rd party dependencies, so pack them together
  DatasetTable: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Datasets/list/DatasetTable.vue'),
  HelpPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/HelpPage.vue'),
  EditUserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/UserProfile/EditUserPage.vue'),
  CreateGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/GroupProfile/CreateGroupPage.vue'),
  ProjectsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Project/ProjectsListPage.vue'),
  SystemHealthPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Admin/SystemHealthPage.vue'),
  GroupsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Admin/GroupsListPage.vue'),
  PrivacyPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PrivacyPage.vue'),
  TermsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/TermsPage.vue'),
  PublicationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PublicationsPage.vue'),

  // These pages use sanitizeHtml, which is big
  ViewGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/GroupProfile/ViewGroupPage.vue'),
  ViewProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Project/ViewProjectPage.vue'),
}
const asyncPages = asyncPagesFreelyTyped as Record<keyof typeof asyncPagesFreelyTyped, AsyncComponent>

const convertLegacyUrls = () => {
  const { pathname, hash, search } = window.location
  if (pathname === '/' && hash && hash.startsWith('#/')) {
    history.replaceState(undefined, undefined as any, hash.slice(1))
  }
  if (pathname === '/annotations') {
    const updatedQueryString = updateDBParam(search.slice(1))
    if (updatedQueryString !== null) {
      history.replaceState(undefined, undefined as any, `${pathname}?${updatedQueryString}`)
    }
  }
}
convertLegacyUrls()

const router = new VueRouter({
  mode: 'history',
  routes: [
    { path: '/(about)?', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
    { path: '/annotations', component: asyncPages.AnnotationsPage },
    {
      path: '/datasets',
      component: DatasetsPage,
      children: [
        { path: '', component: asyncPages.DatasetTable },
        { path: 'summary', component: asyncPages.DatasetSummary },
      ],
    },
    { path: '/datasets/edit/:dataset_id', name: 'edit-metadata', component: asyncPages.MetadataEditPage },
    { path: '/datasets/:dataset_id/add-optical-image', name: 'add-optical-image', component: asyncPages.ImageAlignmentPage },
    { path: '/upload', component: asyncPages.UploadPage },
    { path: '/help', component: asyncPages.HelpPage, meta: { footer: true } },
    { path: '/user/me', component: asyncPages.EditUserPage },

    { path: '/admin/health', component: asyncPages.SystemHealthPage },
    { path: '/admin/groups', component: asyncPages.GroupsListPage },

    { path: '/account/sign-in', component: DialogPage, props: { dialog: 'signIn' } },
    { path: '/account/sign-in-success', redirect: redirectAfterSignIn },
    { path: '/account/create-account', component: DialogPage, props: { dialog: 'createAccount' } },
    { path: '/account/forgot-password', component: DialogPage, props: { dialog: 'forgotPassword' } },
    { path: '/account/reset-password', component: ResetPasswordPage },

    { path: '/group/create', component: asyncPages.CreateGroupPage },
    { path: '/group/:groupIdOrSlug', name: 'group', component: asyncPages.ViewGroupPage },

    { path: '/project/:projectIdOrSlug', name: 'project', component: asyncPages.ViewProjectPage },
    { // Legacy URL sent in "request access" emails up until Feb 2019
      path: '/project/:projectIdOrSlug/manage',
      redirect: { path: '/project/:projectIdOrSlug', query: { tab: 'members' } } as RawLocation,
    },
    { path: '/projects', component: asyncPages.ProjectsListPage },

    { path: '/terms', component: asyncPages.TermsPage, meta: { footer: true } },
    { path: '/privacy', component: asyncPages.PrivacyPage, meta: { footer: true } },
    { path: '/publications', component: asyncPages.PublicationsPage, meta: { footer: true } },

    { path: '/design', component: asyncPages.DesignPage, meta: { footer: true, flex: true } },

    { path: '*', component: NotFoundPage, meta: { footer: true, flex: true } },
  ],
  scrollBehavior(to, from, savedPosition) {
    if (to.hash) {
      return { selector: to.hash, offset: { x: 0, y: 64 } } // offset header height
    }
    if (savedPosition) {
      return savedPosition
    }
    if (to.path !== from.path) {
      return { x: 0, y: 0 }
    }
  },
})

const { href } = router.resolve({ name: 'project', params: { projectIdOrSlug: 'REMOVE' } }, undefined, true)
export const PROJECT_URL_PREFIX = location.origin + href.replace('REMOVE', '')

export default router
