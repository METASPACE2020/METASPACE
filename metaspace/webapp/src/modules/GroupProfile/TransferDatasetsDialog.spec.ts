import { mount, Wrapper } from '@vue/test-utils';
import VueRouter from 'vue-router';
import ElementUI from 'element-ui';
import Vuex from 'vuex';
import Vue from 'vue';
import TransferDatasetsDialog from './TransferDatasetsDialog.vue';
import router from '../../router';
import registerMockComponent from '../../../tests/utils/registerMockComponent';
jest.mock('../../components/DatasetItem.vue', () => require('../../../tests/utils/mockComponent')('dataset-item'));

Vue.use(ElementUI);
registerMockComponent('el-dialog');
Vue.use(VueRouter);
Vue.use(Vuex);

describe('TransferDatasetsDialog', () => {
  const mockDatasets = [
    { "id": "2018-06-28_13h23m10s", "name": "Untreated_3_434", "uploadDT": "2018-06-28T13:23:10.837000" },
    { "id": "2018-06-28_13h21m36s", "name": "Dataset 2", "uploadDT": "2018-06-28T13:21:36.973867" },
    { "id": "2018-06-28_12h03m36s", "name": "Dataset 3", "uploadDT": "2018-06-28T12:03:36.409743" },
    { "id": "2018-06-28_13h20m44s", "name": "Dataset 4", "uploadDT": "2018-06-28T13:20:44.736472" },
  ];
  const mockProps = {
    currentUserId: 'current user id',
    groupName: 'Group Name',
    isInvited: true
  };
  [false, true].forEach(hasDatasets => {
    [false, true].forEach(isInvited => {
      it(`should match snapshot (${hasDatasets ? 'datasets to import' : 'no datasets'}, ${isInvited ? 'invited' : 'requesting access'})`, () => {
        const propsData = { ...mockProps, isInvited };
        const wrapper = mount(TransferDatasetsDialog, { router, propsData });
        wrapper.setData({ allDatasets: hasDatasets ? mockDatasets : [] });

        expect(wrapper).toMatchSnapshot();
      });
    })
  });

  it('should call back on success when some datasets are selected', async () => {
    const wrapper = mount(TransferDatasetsDialog, { router, propsData: mockProps });
    wrapper.setData({ allDatasets: mockDatasets });

    wrapper.find(ElementUI.Checkbox).trigger('click');
    wrapper.findAll(ElementUI.Button)
      .filter((b: Wrapper<ElementUI.Button>) => b.props().type === 'primary')
      .at(0)
      .trigger('click');
    await Vue.nextTick();

    expect(wrapper.emitted('accept')).toEqual([
      [mockDatasets.slice(1).map(ds => ds.id)]
    ])
  })

});
