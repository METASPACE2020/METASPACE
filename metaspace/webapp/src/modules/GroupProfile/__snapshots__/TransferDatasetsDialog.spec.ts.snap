// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransferDatasetsDialog should match snapshot (datasets to import, invited) 1`] = `
<mock-el-dialog visible="" append-to-body="" title="Transfer datasets" class="dialog">
  <div>
    <div>
      <p style="margin-top: 0px;">
        You have previously uploaded one or more datasets without a group. Do you want to transfer any of these datasets to Group Name?
      </p>
      <div class="dataset-list">
        <div>
          <label role="checkbox" aria-checked="true" class="el-checkbox is-checked"><span aria-checked="mixed" class="el-checkbox__input is-checked"><span class="el-checkbox__inner"></span>
            <input type="checkbox" aria-hidden="true" class="el-checkbox__original" value="">
            </span><span class="el-checkbox__label">
            Untreated_3_434
            <span class="reset-color">(Submitted 2018-06-28 at 13:23)</span>
            <!---->
            </span>
          </label>
        </div>
        <div>
          <label role="checkbox" aria-checked="true" class="el-checkbox is-checked"><span aria-checked="mixed" class="el-checkbox__input is-checked"><span class="el-checkbox__inner"></span>
            <input type="checkbox" aria-hidden="true" class="el-checkbox__original" value="">
            </span><span class="el-checkbox__label">
            Dataset 2
            <span class="reset-color">(Submitted 2018-06-28 at 13:21)</span>
            <!---->
            </span>
          </label>
        </div>
        <div>
          <label role="checkbox" aria-checked="true" class="el-checkbox is-checked"><span aria-checked="mixed" class="el-checkbox__input is-checked"><span class="el-checkbox__inner"></span>
            <input type="checkbox" aria-hidden="true" class="el-checkbox__original" value="">
            </span><span class="el-checkbox__label">
            Dataset 3
            <span class="reset-color">(Submitted 2018-06-28 at 12:03)</span>
            <!---->
            </span>
          </label>
        </div>
        <div>
          <label role="checkbox" aria-checked="true" class="el-checkbox is-checked"><span aria-checked="mixed" class="el-checkbox__input is-checked"><span class="el-checkbox__inner"></span>
            <input type="checkbox" aria-hidden="true" class="el-checkbox__original" value="">
            </span><span class="el-checkbox__label">
            Dataset 4
            <span class="reset-color">(Submitted 2018-06-28 at 13:20)</span>
            <!---->
            </span>
          </label>
        </div>
      </div>
      <div class="select-buttons">
        <a href="#">Select none</a> <span> | </span>
        <a href="#">Select all</a>
      </div>
      <!---->
      <div class="button-bar">
        <button type="button" class="el-button el-button--default">
          <!---->
          <!----><span>Cancel</span></button>
        <button type="button" class="el-button el-button--primary">
          <!---->
          <!----><span>Join group and transfer 4 datasets</span></button>
      </div>
    </div>
  </div>
</mock-el-dialog>
`;

exports[`TransferDatasetsDialog should match snapshot (datasets to import, requesting access) 1`] = `
<mock-el-dialog visible="" append-to-body="" title="Transfer datasets" class="dialog">
  <div>
    <div>
      <p style="margin-top: 0px;">
        You have previously uploaded one or more datasets without a group. Do you want to transfer any of these datasets to Group Name?
      </p>
      <div class="dataset-list">
        <div>
          <label role="checkbox" aria-checked="true" class="el-checkbox is-checked"><span aria-checked="mixed" class="el-checkbox__input is-checked"><span class="el-checkbox__inner"></span>
            <input type="checkbox" aria-hidden="true" class="el-checkbox__original" value="">
            </span><span class="el-checkbox__label">
            Untreated_3_434
            <span class="reset-color">(Submitted 2018-06-28 at 13:23)</span>
            <!---->
            </span>
          </label>
        </div>
        <div>
          <label role="checkbox" aria-checked="true" class="el-checkbox is-checked"><span aria-checked="mixed" class="el-checkbox__input is-checked"><span class="el-checkbox__inner"></span>
            <input type="checkbox" aria-hidden="true" class="el-checkbox__original" value="">
            </span><span class="el-checkbox__label">
            Dataset 2
            <span class="reset-color">(Submitted 2018-06-28 at 13:21)</span>
            <!---->
            </span>
          </label>
        </div>
        <div>
          <label role="checkbox" aria-checked="true" class="el-checkbox is-checked"><span aria-checked="mixed" class="el-checkbox__input is-checked"><span class="el-checkbox__inner"></span>
            <input type="checkbox" aria-hidden="true" class="el-checkbox__original" value="">
            </span><span class="el-checkbox__label">
            Dataset 3
            <span class="reset-color">(Submitted 2018-06-28 at 12:03)</span>
            <!---->
            </span>
          </label>
        </div>
        <div>
          <label role="checkbox" aria-checked="true" class="el-checkbox is-checked"><span aria-checked="mixed" class="el-checkbox__input is-checked"><span class="el-checkbox__inner"></span>
            <input type="checkbox" aria-hidden="true" class="el-checkbox__original" value="">
            </span><span class="el-checkbox__label">
            Dataset 4
            <span class="reset-color">(Submitted 2018-06-28 at 13:20)</span>
            <!---->
            </span>
          </label>
        </div>
      </div>
      <div class="select-buttons">
        <a href="#">Select none</a> <span> | </span>
        <a href="#">Select all</a>
      </div>
      <p>
        An email will be sent to the group's principal investigator to confirm your access.
      </p>
      <div class="button-bar">
        <button type="button" class="el-button el-button--default">
          <!---->
          <!----><span>Cancel</span></button>
        <button type="button" class="el-button el-button--primary">
          <!---->
          <!----><span>Request access and transfer 4 datasets</span></button>
      </div>
    </div>
  </div>
</mock-el-dialog>
`;

exports[`TransferDatasetsDialog should match snapshot (no datasets, invited) 1`] = `
<mock-el-dialog visible="" append-to-body="" title="Join group" class="dialog">
  <div>
    <div>
      <p style="margin-top: 0px;">
        Are you sure you want to join Group Name?
      </p>
      <div class="button-bar">
        <button type="button" class="el-button el-button--default el-button--small">
          <!---->
          <!----><span>Cancel</span></button>
        <button type="button" class="el-button el-button--primary el-button--small">
          <!---->
          <!----><span>Join group</span></button>
      </div>
    </div>
  </div>
</mock-el-dialog>
`;

exports[`TransferDatasetsDialog should match snapshot (no datasets, requesting access) 1`] = `
<mock-el-dialog visible="" append-to-body="" title="Join group" class="dialog">
  <div>
    <div>
      <p style="margin-top: 0px;">
        An email will be sent to the group's principal investigator to confirm your access.
      </p>
      <div class="button-bar">
        <button type="button" class="el-button el-button--default el-button--small">
          <!---->
          <!----><span>Cancel</span></button>
        <button type="button" class="el-button el-button--primary el-button--small">
          <!---->
          <!----><span>Request access</span></button>
      </div>
    </div>
  </div>
</mock-el-dialog>
`;
